lib/grids_page.dart

```diff
@@ -32,7 +32,7 @@ class GridsPage extends StatelessWidget {
                       children: List.generate(
                           boardsList.length,
                           (index) =>
-                              SelectGrid(grid: index, currentGrid: grid))),
+                              SelectGrid(grid: boardTemplates.keys.elementAt(index), currentGrid: grid))),
```

Comment:
Iterate throgh keys instead of indexes here. This will make the code much shorter.

Reply:
done